broker/cloud_functions/ps_to_gcs/main.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python3
+# -*- coding: UTF-8 -*-
+
+"""This module is intended to be deployed as a
+Google Cloud Function with a Pub/Sub trigger.
+It will listen to the Pub/Sub stream containing the original alert packet
+and store each alert as an Avro file in Cloud Storage.
+See
+https://cloud.google.com/functions/docs/tutorials/pubsub#functions-prepare-environment-python
+for a tutorial.
+"""
+
+from google.cloud import logging
+from google.cloud import storage
+import base64
+import re
+from pathlib import Path
+from tempfile import SpooledTemporaryFile
+import pickle
+import fastavro
+
+
+# connect to the cloud logger
+logging_client = logging.Client()
+log_name = 'ps-to-gcs-cloudfnc'
+logger = logging_client.logger(log_name)
+
+# bucket to store the Avro files
+bucket_name = 'ardent-cycling-243415_ztf_alert_avro_bucket'
+storage_client = storage.Client()
+bucket = storage_client.get_bucket(bucket_name)
+
+class TempAlertFile(SpooledTemporaryFile):
+    """Subclass of SpooledTemporaryFile that is tied into the log
+
+    Log warning is issued when file rolls over onto disk.
+    """
+
+    def rollover(self) -> None:
+        """Move contents of the spooled file from memory onto disk"""
+
+        log.warning(f'Alert size exceeded max memory size: {self._max_size}')
+        super().rollover()
+
+    @property
+    def readable(self):
+        return self._file.readable
+
+    @property
+    def writable(self):
+        return self._file.writable
+
+    @property
+    def seekable(self):  # necessary so that fastavro can write to the file
+        return self._file.seekable
+
+def upload_ztf_bytes_to_bucket(msg, context) -> None:
```

Comment:
Because this was the only way I was able to successfully reformat the schema. This problem of fixing the schema took me a LONG time to figure out and I tried many different methods. There may be a better method than what I came up with, but I couldn't find it.